agent/hibinit-agent

```diff
@@ -39,23 +39,42 @@ IMDS_API_TOKEN_PATH = 'latest/api/token'
 IMDS_SPOT_ACTION_PATH = 'latest/meta-data/hibernation/configured'
 MAX_SWAP_SIZE_OFFSET_ALLOWED = 100 * 1024
 
+# Sigterm_handler Behaviour Modifiers
+CRITICAL_PROCESS_IN_PROGRESS = False
+SHUTDOWN_REQUESTED = False
+
+
 def log(message):
     if log_to_syslog:
         syslog.syslog(message)
 
-def sigterm_handler(signal, frame):
-    #save the state here or do whatever you want
-    log('Process killed cleaning up!')
-    cmd = "swapon -s | cut -f1,1 | grep -w {name}"
-    cmd = cmd.format(name=SWAP_FILE)
-    swapoff = "swapoff {filename}"
-    cmd = check_output(cmd, shell=True)
-    if cmd.strip() == SWAP_FILE:
-        print("Turning off swap for cleanup")
-        swapoff = swapoff.format(filename=SWAP_FILE)
-        check_call(swapoff, shell=True)
-    if os.path.isfile(SWAP_FILE) and os.access(SWAP_FILE, os.R_OK):
-        os.remove(SWAP_FILE)
+
+def sigterm_handler(signum, frame):
+    if CRITICAL_PROCESS_IN_PROGRESS:
+        global SHUTDOWN_REQUESTED
+        SHUTDOWN_REQUESTED = True
+    else:
+        cmd = "swapon -s | cut -f1,1 | grep -w {name}"
```

Comment:
Forgive me , I thought in the design we decided to remove this but maybe I was mistaken. In any case i think it is better to keep this change.

Reply:
Yes, the graceful termination behaviour should be maintained.

The initial design mentioned an edge case where the main process will not exit if the user manually deleted the swap file while the agent was running. 

To fix this, moved the `exit` command out of the if statement to mantain we make a `best effort` to terminate gracefully.

Reply:
Thanks I think i wrongly interpreted the design then. :)